### PR TITLE
Update README.md for AWS usage.

### PR DESCRIPTION
**Pull request by innovia**
The following PR updates the README.md for how to use vault-unsealer with existing vaults / new vaults.

It also describes the set of permissions needed on IAM for using this tool.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,3 +36,199 @@ Flags:
...
+            "Action": [
+                "kms:DescribeKey",
+                "kms:Encrypt",
+                "kms:Decrypt"
+            ],
+            "Resource": [
+                "arn:aws:kms:<region>:<aws-account-id>:key/<key-id>"
+            ]
+        },
+```
```
Comment (simonswine) [inline README.md]: These IAM rules allow every pod access to your vault unseal key. I strongly suggest to use kube2iam to only aloow that for a Role that is associated to the vault-unsealer pods

Comment (innovia) [reply README.md]: sure thanks

**Comment by innovia** [comment]
@simonswine  Thx for the input - you are 100% right I forgot about it :) 
Round done.

**Comment by FraaJad** [comment]
Thanks @innovia, this AWS documentation was very helpful.  I was also able to get it working with a bit less permissive policy which I've attached.

https://gist.github.com/FraaJad/414821a42c2a7675ab14c1a95d8738b7

**Comment by innovia** [comment]
@simonswine round done. Can we proceed with the PR?

File: README.md [hunk 2]
```diff
@@ -36,3 +36,211 @@ Flags:
...
+  echo $UNSEAL3 > /tmp/vault/unseal3
+  echo $UNSEAL4 > /tmp/vault/unseal4
+
+  echo "Encrypting Vault root token"
+  aws kms encrypt --key-id $KMS_KEY_ID \
+    --plaintext fileb:///tmp/vault/root-key \
+    --output text \
+    --query CiphertextBlob > /tmp/vault/root.enc
+ 
+  echo "Creting a new SSM paramter key ${PREFIX}vault-root for Vault root token"
```
Comment (apenney) [inline README.md]: *Creating

File: README.md [hunk 3]
```diff
@@ -36,3 +36,211 @@ Flags:
...
+
+  for i in {0..4}; do
+    echo "Encrypting unseal${i} key"
+    aws kms encrypt --key-id $KMS_KEY_ID \
+      --encryption-context "Tool=vault-unsealer" \
+      --plaintext fileb:///tmp/vault/unseal${i} \
+      --output text \
+      --query CiphertextBlob > /tmp/vault/unseal${i}.enc
+
+    echo "Creting a new SSM paramter key ${PREFIX}vault-unseal-${i}"
```
Comment (apenney) [inline README.md]: * Creating

File: README.md (hunk 2 above)
Comment (innovia) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (innovia) [reply README.md]: Fixed

**Comment by innovia** [comment]
@apenney 

Thanks for taking a look, I have moved the AWS instructions under docs to a separated file in order to  keep the readme more concise and leave room for future google instructions.

Ami

**Comment by innovia** [comment]
@apenney can we continue with the PR?

**Comment by innovia** [comment]
Can someone continue please with this PR?

**Comment by innovia** [comment]
@simonswine @apenney  ping

File: README.md [hunk 4]
```diff
@@ -36,3 +36,28 @@ Flags:
...
+```
+
+```bash
+cd vault-unsealer
+export GOPATH=`pwd`
+go get github.com/jetstack/vault-unsealer/cmd
+export CI_COMMIT_TAG=<version>
+export CI_COMMIT_SHA=$(git rev-parse HEAD)
+make build
+```
```
Comment (simonswine) [inline README.md]: To be honest not too sure why you would need  more than that(in a recent go version 1.8+):

```
go get github.com/jetstack/vault-unsealer
make -C $(go env GOPATH)/src/github.com/jetstack/vault-unsealer build
```

**Comment by simonswine** [comment]
@innovia yeah sorry but there way much work than time in a day. Just continue nudging. I guess it would have made it easier with smaller pieces. 

Maybe take a look if my build from source instructions work for you

File: README.md (hunk 4 above)
Comment (innovia) [reply README.md]: You're right it was go 1.6 at the time so I'm gonna update the readme build section

your instructions works

thanks!

Comment (innovia) [reply README.md]: @simonswine round++

File: README.md [hunk 5]
```diff
@@ -36,3 +36,24 @@ Flags:
...
+## Build from source
+
+```bash
+git clone https://github.com/jetstack/vault-unsealer.git
+```
+
+```bash
+go get github.com/jetstack/vault-unsealerc
+make -C $(go env GOPATH)/src/github.com/jetstack/vault-unsealer build
+```
```
Comment (simonswine) [inline README.md]: go get is also doing the clone from the code block just before

Also there is now a small typo:
`go get github.com/jetstack/vault-unsealerc` -> `go get github.com/jetstack/vault-unsealer`

Comment (innovia) [reply README.md]: cool yes i missed that section...

anyhow fixed round++

**Pull request merged**